lib/kino/rpc.ex

```diff
@@ -0,0 +1,133 @@
+defmodule Kino.RPC do
+  @moduledoc """
+  Functions for working with remote nodes.
+  """
+
+  @relative_file Path.relative_to_cwd(__ENV__.file)
+
+  @doc """
+  Evaluates the contents given by `string` on the given `node`.
+
+  Returns the value returned from evaluation.
+
+  The code is analyzed for variable references, they are automatically
+  extracted from the caller binding and passed to the evaluation. This
+  means that the evaluated string actually has closure semantics.
+
+  The code is parsed and expanded on the remote node. Also, errors
+  and exists are captured and propagated to the caller.
+
+  See `Code.eval_string/3` for available `opts`.
+  """
+  defmacro eval_string(node, string, opts \\ []) do
+    unless is_binary(string) do
+      raise ArgumentError, "Kino.RPC.eval_string/3 expects a string literal as an argument"
+    end
+
+    used_var_names = used_var_names(string, __CALLER__)
+
+    binding = for name <- used_var_names, do: {name, Macro.var(name, nil)}
+
+    quote do
+      Kino.RPC.__remote_eval_string__(
+        unquote(node),
+        unquote(string),
+        unquote(binding),
+        unquote(opts)
+      )
+    end
+  end
+
+  defp used_var_names(string, env) do
+    case silent_string_to_quoted(string) do
```

Comment:
```suggestion
    case Code.string_to_quoted(string, emit_warnings: false, warn_on_unnecessary_quotes: false) do
```

This will be enough to silence it.

Reply:
We can add a TODO to keep only emit_warnings on v1.16+